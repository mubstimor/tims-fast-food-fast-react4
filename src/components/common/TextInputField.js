import React from 'react';
import PropTypes from 'prop-types';

const TextInputField = ({
  name,
  label,
  type,
  placeholder,
  classname,
  icon
}) => {
  return (
    <div className="form-group">
      <div className="input-group">
        <span className="input-group-addon">
          <i className={icon} />
        </span>
        <input
          name={name}
          label={label}
          type={type}
          placeholder={placeholder}
          className={classname}
          icon={icon}
        />
      </div>
    </div>
  );
};

TextInputField.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  type: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  icon: PropTypes.string,
  classname: PropTypes.string.isRequired
};
TextInputField.defaultProps = {
  label: '',
  placeholder: '',
  icon: ''
};

export default TextInputField;
