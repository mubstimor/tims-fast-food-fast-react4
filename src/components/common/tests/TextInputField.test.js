import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import TextInputField from '../TextInputField';

describe('Login', () => {
  let wrapper;
  const props = {
    name: 'email',
    label: 'email',
    type: 'text',
    placeholder: 'Email',
    classname: 'form-control form-control-lg',
    icon: 'fas fa-envelope',
  };

  it('Should render a text input field', () => {
    console.log(props);
    wrapper = shallow(<TextInputField {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
