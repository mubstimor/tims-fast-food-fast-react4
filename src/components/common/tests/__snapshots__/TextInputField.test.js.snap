// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Should render a text input field 1`] = `ShallowWrapper {}`;
